tidy3d/plugins/autograd/invdes/parametrizations.py

```diff
@@ -1,46 +1,124 @@
-from typing import Tuple
+from __future__ import annotations
 
-import autograd.numpy as np
+from typing import Tuple, Union
+
+import pydantic.v1 as pd
+from numpy.typing import NDArray
+
+from tidy3d.components.base import Tidy3dBaseModel
 
 from ..types import KernelType, PaddingType
-from .filters import make_filter
+from .filters import AbstractFilter, CircularFilter, ConicFilter
 from .projections import tanh_projection
 
 
+class FilterAndProject(Tidy3dBaseModel):
+    """A class that combines filtering and projection operations.
+
+    Parameters
+    ----------
+    filter : AbstractFilter
+        The filter to apply.
+    beta : float = 1.0
+        The beta parameter for the tanh projection.
+    eta : float = 0.5
+        The eta parameter for the tanh projection.
+    """
+
+    filter: AbstractFilter
+    beta: pd.NonNegativeFloat = 1.0
+    eta: pd.NonNegativeFloat = 0.5
+
+    def __call__(self, array: NDArray, beta: float = None, eta: float = None) -> NDArray:
+        """Apply the filter and projection to an input array.
+
+        Parameters
+        ----------
+        array : NDArray
+            The input array to filter and project.
+        beta : float = None
+            The beta parameter for the tanh projection. If None, uses the instance's beta.
+        eta : float = None
+            The eta parameter for the tanh projection. If None, uses the instance's eta.
+
+        Returns
+        -------
+        NDArray
+            The filtered and projected array.
+        """
+        filtered = self.filter(array)
+        beta = beta if beta is not None else self.beta
+        eta = eta if eta is not None else self.eta
+        projected = tanh_projection(filtered, beta, eta)
+        return projected
+
+    @classmethod
+    def from_params(
+        cls,
+        radius: Union[float, Tuple[float, ...]] = None,
+        dl: Union[float, Tuple[float, ...]] = None,
+        *,
+        size_px: Union[int, Tuple[int, ...]] = None,
+        beta: float = 1.0,
+        eta: float = 0.5,
+        filter_type: KernelType = "conic",
+        padding: PaddingType = "reflect",
+    ) -> FilterAndProject:
+        """Create a FilterAndProject instance from parameters.
```

Comment:
backslash for `FilterAndProject`. 

What parameters? maybe clarify "from parameters for both ``make_filter`` and ``tanh_projection``"

Reply:
I'm pretty unhappy with the parameters and the `from_params()` constructor actually...
The problem is that in principle, the `from_params()` will just need to contain all arguments that are necessary to construct the classes that are used within the parametrization. Same goes for the penalty class. So it's a sort of catch-all method that really doesn't make it much simpler just constructing a Filter yourself and passing it to the default constructor. Not sure how to best solve that tbh :/

Reply:
not sure if this makes sense, but could try `FilterAndProject` just taking all the parameters as fields, and then having a `filter` property?  I might be confused though.